Assets/Scripts/Mixed/GameObjects/SharedMaterials.cs

```diff
@@ -48,14 +62,45 @@ public Material GetMaterialById(int id)
 
         public int GetIdForMaterial(Material material)
         {
-            foreach (var key in materialDictionary.Keys)
-            {
-                if (materialDictionary[key] == material)
-                {
-                    return key;
+            var matName = material.name;
+            if(materialKeyByNameDictionary.TryGetValue(matName, out var materialId)){
+                return materialId;
+            }
+
+            throw new System.Exception($"Given material {material.name} was not found on the material library. Make sure to Add the material to the Scriptable Object Material library on scene.");
+        }
+
+        public void Convert(Entity entity, EntityManager dstManager, GameObjectConversionSystem conversionSystem)
+        {
+            BlobAssetReference<SharedMaterialBlobAsset> sharedMaterialBlobAssetReference;
+            using(var blobBuilder = new BlobBuilder(Unity.Collections.Allocator.Temp)){
+                ref var blobAsset = ref blobBuilder.ConstructRoot<SharedMaterialBlobAsset>();
+                var sharedMaterialArray = blobBuilder.Allocate(ref blobAsset.Materials, materialDictionary.Count);
+
+                foreach(var dicVal in materialDictionary){
+                    sharedMaterialArray[dicVal.Key] = new SharedMaterial{Value = dicVal.Value};
                 }
+
+                sharedMaterialBlobAssetReference = blobBuilder.CreateBlobAssetReference<SharedMaterialBlobAsset>(Unity.Collections.Allocator.Persistent);
             }
-            throw new System.Exception($"Given material was not found on the material library. Make sure to Add the material to the Scriptable Object Material library on scene.");
+
+            dstManager.AddComponentData(entity, new SharedMaterialData{
+                sharedMaterialsBlobAssetRef = sharedMaterialBlobAssetReference
+            });
         }
     }
+
+    public struct SharedMaterial
+    {
+        public Material Value;
+    }
+
+    public struct SharedMaterialBlobAsset
+    {
+        public BlobArray<SharedMaterial> Materials;
```

Comment:
I like this, is it possible do this with prefabs (entities) instead of just an asset from an entity?